plotly/tools.py

```diff
@@ -1477,339 +1767,155 @@ def create_scatterplotmatrix(df, useindex = False, index = " ",
+        from plotly.tools import FigureFactory as FF
+
+        import numpy as np
+        import pandas as pd
+
+        # Create 4X4 dataframe with index
+        df = pd.DataFrame(np.random.randn(10, 4),
+                           columns=["A", "B", "C", "D"])
         
-            # Check if length of col_names is equal to the
-            # number of lists in matrix
-            if len(col_names) != dim:
-                raise exceptions.PlotlyError("Your list of variable_"
-                                             "names must match the "
-                                             "number of lists in your "
-                                             "array. That is to say that "
-                                             "both lists must have the "
-                                             "same dimension.")
-
-            # Insert col_names into the figure
-            for j in range(dim):
-                xaxis_place = "xaxis" + str(dim*dim - dim + 1 + j)
-                fig['layout'][xaxis_place].update(title = col_names[j])
-
-            for j in range(dim):
-                yaxis_place = "yaxis" + str(1 + dim*j)
-                fig['layout'][yaxis_place].update(title = col_names[j])
-
-            # Set height and width if not already selected by user
-            if (height == " "):
-                height = 400 + 200*(dim - 1)
-
-            if (width == " "):
-                width = 400 + 200*(dim - 1)
-
-            if diagonal == "Histogram":
-                fig['layout'].update(
-                    height = height, width = width,
-                    title = title,
-                    showlegend = True,
-                    barmode = "stack")
-                return fig
-
-            if diagonal == "Scatter":
-                fig['layout'].update(
-                    height = height, width = width,
-                    title = title,
-                    showlegend = True)
-                return fig
+        df["Fruit"] = pd.Series(["apple","apple","pear","apple","apple",
+                                 "pear","pear","pear","apple","pear"])
+
+        # Create scatterplot matrix
+        fig = FF.create_scatterplotmatrix(df, useindex = True, index = "Fruit",
+                                          symbol = 2, size = 8,
+                                          diag = "Histogram")
+
+        # Plot
+        fig = py.iplot(fig, filename='ScatterPlot Matrix')
+        ```
+        """
+        # TODO: protected until #282
+        matrix = []
+        headers = []
+        index_vals = []
+        global diag_choices
+        diag_choices = ["Scatter", "Histogram", "box"]
+        colors = ["rgb(31, 119, 180)", "rgb(255, 127, 14)",
+                  "rgb(44, 160, 44)", "rgb(214, 39, 40)",
+                  "rgb(148, 103, 189)", "rgb(140, 86, 75)",
+                  "rgb(227, 119, 194)", "rgb(127, 127, 127)",
+                  "rgb(188, 189, 34)", "rgb(23, 190, 207)"]
```

Comment:
You can do the same with this `colors` declaration.

`DEFAULT_SCATTER_MATRIX_COLORS = ..`